sktime/forecasting/hf_momentfm_forecaster.py

```diff
@@ -0,0 +1,778 @@
+            pct_start=self.pct_start,
+        )
+
+        # Gradient clipping value
+        max_norm = self.max_norm
+
+        self.model, optimizer, train_dataloader, val_dataloader, scheduler = (
+            accelerator.prepare(
+                self.model, optimizer, train_dataloader, val_dataloader, scheduler
+            )
+        )
+
+        while cur_epoch < max_epoch:
+            cur_epoch = _run_epoch(
+                cur_epoch,
+                accelerator,
+                criterion,
+                optimizer,
+                scheduler,
+                self.model,
+                max_norm,
+                train_dataloader,
+                val_dataloader,
+            )
+
+        if self.return_model_to_cpu:
+            self.model.to("cpu")
+            empty_cache()
+
+        return self
+
+    def _predict(self, y, X=None, fh=None):
+        """Predict method to forecast timesteps into the future.
+
+        fh should not be passed here and
+        must be the same length as the one used to fit the model.
+        """
+        # use y values from fit if y is None in predict
+        if y is None:
+            y = self._y
+
+        index = self._fh.to_absolute_index(self.cutoff)
+        from torch import from_numpy
+
+        self.model = self.model.to(self._device)
+        self.model.eval()
+        y_index_names = list(y.index.names)
+        if isinstance(y.index, pd.MultiIndex):
+            y_ = _frame2numpy(y)
+        else:
+            y_ = np.expand_dims(y.values, axis=0)
+
+        num_instances, sequence_length, num_channels = (
+            y_.shape
+        )  # shape of our input to predict
+        # raise warning if sequence length of y is greater than the sequence
+        # length used to fit the model
+        if sequence_length > self._seq_len:
+            warnings.warn(
+                f"Sequence length of {sequence_length} was found which is greater "
+                "than sequence "
+                f"length {self._seq_len} used to fit the model. The most recent"
+                f" {self._seq_len} values will be used.",
+                stacklevel=2,
+            )
+            # truncate code
+            # only retain the most recent self._seq_len values if greater than
+            # self._seq_len
+            y_ = y_[:, -self._seq_len :, :]
+            sequence_length = self._seq_len
+        if sequence_length < self._moment_seq_len:
+            # if smaller, need to pad values
+            y_ = _create_padding(
+                y_,
+                (num_instances, self._moment_seq_len - sequence_length, num_channels),
+            )
+            input_mask = _create_mask(
+                sequence_length, self._moment_seq_len - sequence_length
+            )
+        else:  # this means sequence_length = self._seq_len == 512
+            input_mask = _create_mask(self._moment_seq_len)
+        if num_channels != self._y_shape[1]:
+            raise ValueError(
```

Comment:
You are inheriting from `BaseGlobalForecaster`. How do you control multivariate forecasting? 

I haven't seen a mechanism to switch from Global to Multivariate forecasting in the code.